### PR TITLE
Added ZPopMax and ZPopMin to miniredis.

### PR DESCRIPTION
**Pull request by rafalmnich**

### CODE REVIEW

File: cmd_sorted_set_test.go [hunk 1]
```diff
@@ -1416,3 +1416,139 @@ func TestSSRange(t *testing.T) {
...
+		_, err = redis.String(c.Do("ZPOPMAX", "set", 1, "toomany"))
+		assert(t, err != nil, "ZPOPMAX error")
+		// Wrong type of key
+		s.Set("str", "value")
+		_, err = redis.Int(c.Do("ZPOPMAX", "str", 1, 2))
+		assert(t, err != nil, "ZPOPMAX error")
+	}
+}
+
+// Test ZPOPMAX
```
Comment (alicebob) [inline cmd_sorted_set_test.go]: s/MAX/MIN/ :)

**Comment by alicebob** [comment]
Thanks for the PR! It looks good, but could you add some tests to the ./integration/ subdir?

I'll see if I can get travis to use redis 5.0 in the meantime...

**Comment by alicebob** [comment]
> I'll see if I can get travis to use redis 5.0 in the meantime...

done, that works now.

**Comment by joeirimpan** [comment]
Is there anything left for this patch to get merged?

**Comment by alicebob** [comment]
hi @joeirimpan ,

yes, it still needs some tests in the `./integration/` subdir, where the output is compared against a real Redis.

**Comment by joeirimpan** [comment]
@rafalmnich Do you mind if I raise a PR fixing and adding integration tests?

File: cmd_sorted_set_test.go [hunk 2]
```diff
@@ -1416,3 +1416,139 @@ func TestSSRange(t *testing.T) {
...
+	s.ZAdd("z", 3, "three")
+	s.ZAdd("z", 3, "drei")
+	s.ZAdd("z", math.Inf(+1), "inf")
+
+	{
+		n, err := redis.Strings(c.Do("ZPOPMAX", "z", 2))
+		ok(t, err)
+		equals(t, 4, len(n))
+
+		equals(t, []string{"one", "1", "two", "2"}, n)
```
Comment (rikonor) [inline cmd_sorted_set_test.go]: I think you may have gotten this backwards.
```
redis:6379> ZRANGE z 0 -1 WITHSCORES
1) "one"
2) "1"
3) "two"
4) "2"
5) "three"
6) "3"
7) "four"
8) "4"

redis:6379> ZPOPMAX z 2
1) "four"
2) "4"
3) "three"
4) "3"
```
So you basically mixed up your ZPOPMIN and ZPOPMAX implementations.

Comment (joeirimpan) [reply cmd_sorted_set_test.go]: yes. there is a mixup.

**Comment by joeirimpan** [comment]
@alicebob Can I take over this patch?
https://github.com/joeirimpan/miniredis/commits/fix/zpopx fixes this.

**Comment by alicebob** [comment]
> @alicebob Can I take over this patch?
> https://github.com/joeirimpan/miniredis/commits/fix/zpopx fixes this.

Sure! Could you make a new PR, please?

**Comment by joeirimpan** [comment]
@alicebob here it is https://github.com/alicebob/miniredis/pull/72

**Comment by alicebob** [comment]
See the linked PR. Merged into master. Thanks!

**Pull request closed without merging**